zkevm-circuits/src/witness/block.rs

```diff
@@ -130,6 +135,108 @@ impl<F: Field> Block<F> {
     pub(crate) fn get_big_modexp(&self) -> Vec<BigModExp> {
         self.precompile_events.get_modexp_events()
     }
+
+    pub(crate) fn get_num_rows_required_no_padding(&self) -> usize {
+        let mut num_rows = 0;
+        let mut counter = HashMap::new();
+        let mut step_num = 0;
+        for transaction in &self.txs {
+            step_num += transaction.steps.len();
+            for step in &transaction.steps {
+                let height = step.execution_state.get_step_height();
+                num_rows += height;
+                *counter.entry(step.execution_state).or_insert(0) += height;
+            }
+        }
+        // TODO: change to log::trace?
+        let mut counter_vec: Vec<_> = counter.into_iter().collect();
+        // DESC
+        counter_vec.sort_by_key(|(_e, c)| -(*c as i64));
+        let print_top_k = 10;
+        for (idx, (e, usage)) in counter_vec.iter().enumerate() {
+            if idx > print_top_k {
+                break;
+            }
+            let height = e.get_step_height();
+            log::debug!(
+                "evm circuit row usage: {:?}, step ratio {}, row ratio {}, height {}",
+                e,
+                (usage / height) as f64 / step_num as f64,
+                *usage as f64 / num_rows as f64,
+                height
+            );
+        }
+
+        // EndBlock
```

Comment:
remove line170-line175

Reply:
Also removed in commit [4aa4e81](https://github.com/scroll-tech/zkevm-circuits/pull/933/commits/4aa4e81892a065ce170436fb2d32115f70227570).